### PR TITLE
89 add data elements to descriptions

### PR DESCRIPTION
**Pull request by ibevers**

### CODE REVIEW

**Comment by ibevers** [comment]
@satra @alistairewj I did this based on the ReproSchema. Many data element ids don't correspond to ReproSchema. Currently left to do:
- [ ] Add URLs for data elements
- [ ] Add URLs for assessments
- [ ] Add descriptions for assessments
- [ ] Add assessment names
- [ ] Decide how to address elements with ids that don't correspond to ReproSchema.

**Comment by ibevers** [comment]
Some phenotype json entry names are formatted as [item name]___[response choice]. In these cases, it appears that there are separate entries for each response choice. For example:
```
    "diagnosis_ld_assessment___aural_perceptual_evaluation": {
        "description": "Indicates if the participant underwent an aural-perceptual evaluation as part of their laryngeal dystonia assessment."
    },
    "diagnosis_ld_assessment___laryngoscopy_or_stroboscopy": {
        "description": "Indicates if the participant underwent laryngoscopy or stroboscopy as part of their laryngeal dystonia assessment."
    },
    "diagnosis_ld_assessment___neurological_evaluation": {
        "description": "Indicates if the participant underwent a neurological evaluation as part of their laryngeal dystonia assessment."
    },
```

**Comment by ibevers** [comment]
Additionally, there are some that match the items in b2ai_redcap2rs, except they are missing a `ef_` prefix, for example, in `phenotype/participant.json` we have:
```
    "laryng_cancer": {
        "description": "Indicates if the participant has laryngeal cancer."
    },
    "benign_cord_lesion": {
        "description": "Indicates if the participant has benign lesions of the vocal cord such as nodules, polyps, or cysts."
    }
```

**Comment by ibevers** [comment]
> Some phenotype json entry names are formatted as [item name]___[response choice]. In these cases, it appears that there are separate entries for each response choice. For example:
> 
> ```
>     "diagnosis_ld_assessment___aural_perceptual_evaluation": {
>         "description": "Indicates if the participant underwent an aural-perceptual evaluation as part of their laryngeal dystonia assessment."
>     },
>     "diagnosis_ld_assessment___laryngoscopy_or_stroboscopy": {
>         "description": "Indicates if the participant underwent laryngoscopy or stroboscopy as part of their laryngeal dystonia assessment."
>     },
>     "diagnosis_ld_assessment___neurological_evaluation": {
>         "description": "Indicates if the participant underwent a neurological evaluation as part of their laryngeal dystonia assessment."
>     },
> ```

These items are columns in the RedCap data dictionary. They are also in the participant.json files. Possible paths forward:
1. Replace them with the structure in `b2ai_redcap2rs`. This means the entries won't map neatly to the individual participant.json files. Shows the structure clearly.
2. Add an entry for the overall item. This might duplicate some information. Duplication could be avoided by moving the types to the individual items.

2 is probably safer, so I will go with that.

**Comment by ibevers** [comment]
There are no items or activities that correspond to these two. Since there only two, they should be updated manually.
`['q_generic_demographics_complete', 'subjectparticipant_eligible_studies_complete']`

- [ ] Update ['q_generic_demographics_complete', 'subjectparticipant_eligible_studies_complete'] manually

**Comment by alistairewj** [comment]
The structure seems fine to me - where did you grab the info? The protocol?

Side note but it's probably worth a rebase, squash and merge for this PR as there are commits with 2,000,000+ lines which are later removed

**Comment by ibevers** [comment]
@alistairewj I got the info from here:
https://github.com/sensein/b2ai-redcap2rs/tree/main/activities

That's a good suggestion, thanks

**Comment by satra** [comment]
before we squash and merge this, can we check a few things:

1. data element names match with the uber csv file that @alistairewj is producing
2. for choices: the values for choice columns in that file match with the values (not the names) in the data dictionary for those choices.
3. that all data elements in the csv that is produced have corresponding data elements here.

**Comment by ibevers** [comment]
@alistairewj what is the uber csv file that you are making?

**Comment by alistairewj** [comment]
the phenotype.tsv file which simply concatenates all the tsv in phenotype/

Column headers are currently the same as in the phenotype/*.tsv files. I do think I will merge multiple one-hot encoded columns into a single column, e.g. `ef_fluent_languages__1`, `ef_fluent_languages__2`, etc into a single `ef_fluent_languages` column.

**Comment by satra** [comment]
>  do think I will merge multiple one-hot encoded columns into a single column,

perhaps only for elements for which this is a single choice rather than multiple choice.

**Comment by ibevers** [comment]
>for choices: the values for choice columns in that file match with the values (not the names) in the data dictionary for those choices.

Steps:
- [ ] Get list of non-matching response choices.
non_matching
For each phenotype/*.json:
    For each data element:
         Filter data dictionary by "Variable / Field Name" by data element name.
         For each response choice:
             Lexically match with "Choices, Calculations, OR Slider Labels" column in filtered data dictionary.
             If no match:
                  Append to non_matching.
  - [ ] Determine how to fix non-matching choices.

File: src/b2aiprep/prepare/resources/b2ai-data-bids-like-template/phenotype/voiceSeverity.json [hunk 1]
```diff
@@ -1,11 +1,60 @@
...
+                "question": {
+                    "en": "Describe the severity of auditory-perceptual attributes of a voice problem*Overall severity scale (0-100)"
+                },
+                "datatype": [
+                    "xsd:integer"
+                ],
+                "choices": [
+                    {
+                        "name": {
+                            "en": "0 "
```
Comment (satra) [inline src/b2aiprep/prepare/resources/b2ai-data-bids-like-template/phenotype/voiceSeverity.json:39]: can we check why these have spaces?

File: src/b2aiprep/prepare/resources/b2ai-data-bids-like-template/phenotype/gad7.json [hunk 2]
```diff
@@ -1,29 +1,307 @@
...
+                "question": {
+                    "en": "Becoming easily annoyed or irritable."
+                },
+                "datatype": [
+                    "xsd:integer"
+                ],
+                "choices": [
+                    {
+                        "name": {
+                            "en": "Not at all "
```
[the remark this answers is not in the document]
Comment (alistairewj) [reply src/b2aiprep/prepare/resources/b2ai-data-bids-like-template/phenotype/gad7.json:208]: I'm fairly sure they have spaces in redcap

[the remark this answers is not in the document]
Comment (ibevers) [reply src/b2aiprep/prepare/resources/b2ai-data-bids-like-template/phenotype/gad7.json:208]: I will check the data dictionary

**Comment by ibevers** [comment]
> > for choices: the values for choice columns in that file match with the values (not the names) in the data dictionary for those choices.
> 
> Steps:
> 
> * [ ]  Get list of non-matching response choices.
> * non_matching
> * For each phenotype/*.json:
>   
>   * For each data element:
>     
>     * Filter data dictionary by "Variable / Field Name" by data element name.
>     * For each response choice:
>       
>       * Lexically match with "Choices, Calculations, OR Slider Labels" column in filtered data dictionary.
>       * If no match:
>         Append to non_matching.
> * [ ]  Determine how to fix non-matching choices.

Decided not to do since Evan checked the JSONs with the TSVs.

**Pull request merged**